src/main/java/com/mraof/minestuck/world/gen/feature/tree/TreeStumpFeature.java

```diff
@@ -0,0 +1,66 @@
+package com.mraof.minestuck.world.gen.feature.tree;
+
+import com.mojang.serialization.Codec;
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.world.gen.feature.AbstractTemplateFeature;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Vec3i;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.WorldGenLevel;
+import net.minecraft.world.level.levelgen.Heightmap;
+import net.minecraft.world.level.levelgen.feature.FeaturePlaceContext;
+import net.minecraft.world.level.levelgen.feature.configurations.NoneFeatureConfiguration;
+
+/**
+ * Structure nbt defined tree stumps intended for Forest Lands
+ */
+public class TreeStumpFeature extends AbstractTemplateFeature<NoneFeatureConfiguration>
+{
+	private static final ResourceLocation TREE_STUMP = new ResourceLocation(Minestuck.MOD_ID, "tree_stump");
+	
+	public TreeStumpFeature(Codec<NoneFeatureConfiguration> codec)
+	{
+		super(codec);
+	}
+	
+	@Override
+	protected ResourceLocation pickTemplate(RandomSource random)
+	{
+		return TREE_STUMP;
+	}
+	
+	@Override
+	public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context)
+	{
+		//continue with normal placement only if the terrain is relatively even
+		if(isEvenTerrain(context))
+			return super.place(context);
+		else
+			return false;
+	}
+	
+	private boolean isEvenTerrain(FeaturePlaceContext<NoneFeatureConfiguration> context)
+	{
+		WorldGenLevel level = context.level();
+		BlockPos pos = context.origin();
+		int minY = Integer.MAX_VALUE;
+		int maxY = Integer.MIN_VALUE;
+		
+		for(BlockPos floorPos : BlockPos.betweenClosed(pos, pos.offset(8, 0, 8)))
```

Comment:
> `BlockPos.betweenClosed(pos, pos.offset(8, 0, 8))`

I'm guessing that this is supposed to match up with the space of the placed structure. Do you know if it actually matches up?

Reply:
I cant actually confirm with certainty, but I based it off of minWorldHeightInSize. If it isnt taking rotation into effect, it still seems to do the job it was intended to do

Reply:
Took a closer look at `AbstractTemplateFeature`, and yeah it does seem to match it correctly. But apparently it's not supposed to. The position used for placing the template and which is passed into `pickY()` and `minWorldHeightInSize()` is supposed to be offset so that the template is centered on the origin, but due to a mistake in `AbstractTemplateFeature.place()`, that position actually ended up just being the origin pos, which places the template with a corner on the origin and makes the area used here match up with the template area on accident.

So that's a separate issue to be fixed in `AbstractTemplateFeature`. But more importantly for this PR, I think that code in this class shouldn't be dependent on how the template is placed in `AbstractTemplateFeature`, which means some sort of changes here.

I do have an idea I'd like to try out for that, and I think I would like to do it in my own PR. With that in mind, I am fine merging this without any additional changes in this PR.